doc/Overview/Mvux/Advanced/Commands.md

```diff
@@ -0,0 +1,209 @@
+
+Commands can be bound to UI elements. To do this, the `Command` property of the UI element can be used. For example, the following code binds the `Click` event of a button to the `Save` command:
+
+```
+<Button Command="Save">Save</Button>
+```
+
+## Asynchronous commands
+
+The MVUX [`IAsyncCommand`](https://github.com/unoplatform/uno.extensions/blob/main/src/Uno.Extensions.Reactive/Presentation/Commands/IAsyncCommand.cs) interface, is a Command that implements [`ICommand`](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.input.icommand?view=windows-app-sdk-1.3) and adds support for asynchronous operations.  
+As it implements `ICommand`, it can be bound from the View as a regular Command (e.g. in a `Button.Command` property). The advantage is that it can be invoked asynchronously.  
+In addition, it also implements [`INotifyPropertyChanged`](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.inotifypropertychanged), to enable tracking if its properties have changed, and [`ILoadable`](xref:Toolkit.Controls.LoadingView#iloadable) - an Uno interface that provides information of an object's state whether it's currently in execution mode or not, and notifies subscribers when this state changes.
+
+![A class diagram of System.ComponentModel.ICommand inheritence structure](../Assets/Commands-2.jpg)
+
+## Creating commands
+
+There are several methods of how to create an MVUX command.
+
+1. Using code generation
+
+    ### Basic commands
+
+    When creating a method in the Model, it will be generated as an `AsyncCommand` in the Bindable Model.
+
+    For example, when declaring a method of the following signatures:
+
+    ```csharp
+    public void DoWork();
+    public ValueTask DoWork();
+    public ValueTask DoWork(CancellationToken ct);    
+    ```
+
+    A `DoWork` command will be generated in the Bindable Proxy Model. When that command is executed via a button-click etc., the method in the Model will be called:
+
+    ```xml
+    <Button Command="{Binding DoWork}" />
+    ```
+
+    As you may have noticed, the method can be asynchronous and take a `CancellationToken`, but these are not mandatory and commands will be generated for synchronous methods as well.
+
+    ### Adding command parameters
+
+    An additional parameter can be added to the method, which is then assigned with the value of the `CommandParameter` received from the View. For instance, when using a Button and clicking it, the method will be called with the [`Button.CommandParameter`](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.primitives.buttonbase.commandparameter) value, given that the type of the value matches the method parameter type. Otherwise the command's `CanExecute` will be false thereby disabling the button:
```

Comment:
"An additional parameter can be added to the method, which is then assigned with the value of the `CommandParameter` received from the View."

I think you're trying to say:

If the method takes a parameter, this can be specified in the View by setting the `CommandParameter` property on the element which has the `Command` specified.

---

`, given that the type of the value matches the method parameter type.`

Is this really true? Is it doing Type comparison? Or is it down to the developer to ensure that the value passed is of a suitable type that is supports runtime conversion or uses a suitable converter?

---

"Otherwise the command's `CanExecute` will be false thereby disabling the button:"

Is this really true?
Does the default `CanExecute` method do runtime checking of Types that it maps to the IsEnabled property?

Reply:
If I'm wrong about the above and the behavior described is correct because of a difference to how things work in MVUX (compared to MVVM), then this should be called out.

Reply:
Note that the MVVM experience is a runtime exception due to a XAML parsing failure.

Rather than the button being disabled, it would be more helpful if the MVUX source generators could identify the issue at compile-time. Or, is this behavior intentional so that it can support the binding of different objects as parameters at runtime?